src/ftest.jl

```diff
@@ -0,0 +1,130 @@
+    npreds1 > npreds2 && return false 
+    
+    @inbounds for i in 1:npreds1
+        var_in_mod2 = false
+        for j in 1:npreds2
+            if view(pred1, :, i) == view(pred2, :, j)
+                var_in_mod2 = true
+                break
+            end
+        end
+        if !var_in_mod2
+            # We have found a predictor variable in model 1 that is not in model 2
+            return false 
+        end
+    end
+
+    return true
+end
+
+"""
+    `function ftest(mod::LinPredModel...)`
+Test if mod[i] fits significantly better than mod[i+1]
+
+Note: This function can easily be used to do an ANOVA. ANOVA is nothing more than a test
+to see if one model fits the data better than another.
+
+# Examples:
+As usual, we want to compare a result across two or more treatments. In the classic ANOVA
+framework, our null hypothesis is that Result~1 is a perfectly good fit to the data. 
+The alternative for ANOVA is that Result~Treatment is a better fit to the data than Result~1
+```jldoctest
+julia> d = DataFrame(Treatment=[1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2.], Result=[1.1, 1.2, 1, 2.2, 1.9, 2, .9, 1, 1, 2.2, 2, 2])
+julia> mod = lm(Result~Treatment, d)
+julia> nullmod = lm(Result~1, d)
+julia> ft = ftest(mod.model, nullmod.model)
+====================================================================
+Fisher 2-model F test with 10 and 1 degrees of freedom
+Sums of squared residuals: 0.12833333333333344 and 3.229166666666667
+F* = 241.62337662337643 p = 2.481215056713184e-8
+```
+"""
+function ftest(mods::LinPredModel...)
+    nmodels = length(mods)
+    for i in 2:nmodels
+        @argcheck issubmodel(mods[i], mods[i-1]) "F test is only valid if model 2 is nested in model 1"
+    end
+
+    SSR1 = [deviance(mods[i]) for i in 1:nmodels-1]
+    SSR2 = [deviance(mods[i]) for i in 2:nmodels]
+
+    nparams1 = [dof(mods[i]) for i in 1:nmodels-1]
+    nparams2 = [dof(mods[i]) for i in 2:nmodels]
+
+    df2 = nparams1-nparams2
+    df1 = [Int(dof_residual(mods[i])) for i in 1:nmodels-1]
+
+    MSR1 = (SSR2-SSR1)./df2
+    MSR2 = SSR1./df1
+
+    fstat = MSR1./MSR2
+    pval = ccdf.(FDist.(df2, df1), fstat)
+
+    return FTestTable(SSR1, SSR2, df1, df2, MSR1, MSR2, fstat, pval)
+end
+
+### Utility functions to show FTestResult and MultiFTestResult
+function show(io::IO, x::FTestTable)
+    if get(io, :compact, true)
+#=        if x.ntests == 1
+            lines = [string("Fisher 2-model F test with ", x.df1[1], " and ", x.df2[1], " degrees of freedom"), string("Sums of squared residuals: ", round(x.SSR1[1], 6), " and ", round(x.SSR2[1], 6)), string("F* = ", round(x.fstat[1], 6), " p = ", PValue(x.pval[1]))]
+            outlen = maximum(length, lines)
+
+            println(io, RepString("=", outlen))
+            for line in lines
+                println(io, line)
+            end
+        else=#
+	    nequals = 86
+            println(io, "Fisher 2-model F tests (between ajdacent pairs of models)")
```

Comment:
I didn't even realize that Fisher was affiliated with it; I thought it was Snedecor. I think it's most common to just call it an F test.